airbyte-integrations/connectors/source-marketo/source_marketo/source.py

```diff
@@ -0,0 +1,530 @@
+from airbyte_cdk.sources.streams.http import HttpStream
+from airbyte_cdk.sources.streams.http.auth import Oauth2Authenticator
+
+from .utils import STRING_TYPES, clean_string, format_value, to_datetime_str
+
+
+class MarketoStream(HttpStream, ABC):
+    primary_key = "id"
+    data_field = "result"
+    page_size = 300
+
+    def __init__(self, config: Mapping[str, Any], stream_name: str = None, param: Mapping[str, Any] = None, export_id: int = None):
+        super().__init__(authenticator=config["authenticator"])
+        self.config = config
+        self.start_date = config["start_date"]
+        self.window_in_days = config["window_in_days"]
+        self._url_base = config["base_url"]
+        self.stream_name = stream_name
+        self.param = param
+        self.export_id = export_id
+
+    @property
+    def url_base(self) -> str:
+        return self._url_base
+
+    def path(self, **kwargs) -> str:
+        return f"/rest/v1/{self.name}.json"
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        next_page = response.json().get("nextPageToken")
+
+        if next_page:
+            return {"nextPageToken": next_page}
+
+    def request_params(
+        self,
+        stream_state: Mapping[str, Any],
+        stream_slice: Mapping[str, Any] = None,
+        next_page_token: Mapping[str, Any] = None,
+    ) -> MutableMapping[str, Any]:
+        params = {"batchSize": self.page_size}
+        if next_page_token:
+            params.update(**next_page_token)
+        return params
+
+    def parse_response(self, response: requests.Response, stream_state: Mapping[str, Any], **kwargs) -> Iterable[Mapping]:
+        data = response.json().get(self.data_field, [])
+
+        for record in data:
+            yield record
+
+
+class IncrementalMarketoStream(MarketoStream):
+    cursor_field = "createdAt"
+
+    def filter_by_state(self, stream_state: Mapping[str, Any] = None, record: Mapping[str, Any] = None) -> Iterable:
+        """
+        Endpoint does not provide query filtering params, but they provide us
+        cursor field in most cases, so we used that as incremental filtering
+        during the parsing.
+        """
+
+        if not stream_state or record[self.cursor_field] >= stream_state.get(self.cursor_field):
+            yield record
+
+    def parse_response(self, response: requests.Response, stream_state: Mapping[str, Any], **kwargs) -> Iterable[Mapping]:
+        json_response = response.json().get(self.data_field) or []
+
+        for record in json_response:
+            yield from self.filter_by_state(stream_state=stream_state, record=record)
+
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
+        return {
+            self.cursor_field: max(latest_record.get(self.cursor_field, ""), current_stream_state.get(self.cursor_field, self.start_date))
```

Comment:
We can use the `self.start_date` instead of `""` in `latest_record.get(self.cursor_field, "")`